File: 1630-count-odd-numbers-in-an-interval-range/count-odd-numbers-in-an-interval-range.java
class Solution {
    public int countOdds(int low, int high) {
        int oddCountTillHigh = (high + 1) / 2;
        int oddCountTillLow = (low + 1) / 2;
        int oddCount = oddCountTillHigh - oddCountTillLow;
        oddCount += low % 2;
        return oddCount;
    }
}

File: 1630-count-odd-numbers-in-an-interval-range/count-odd-numbers-in-an-interval-range.py
class Solution:
    def countOdds(self, low: int, high: int) -> int:
        odd_count_till_high = (high + 1) // 2
        odd_count_till_low = (low + 1) // 2
        odd_count = odd_count_till_high - odd_count_till_low
        odd_count += low % 2
        return odd_count